From adbc26d972d3a9a6f90945a272dd4eaf1a03fedc Mon Sep 17 00:00:00 2001
From: Jiang Lu <lu.jiang@windriver.com>
Date: Tue, 18 Feb 2014 16:50:41 +0800
Subject: [PATCH] fsl_dpaa:bman:call sysfs_attr_init() for dynamically
 allocated sysfs attrs

Dynamically allocated sysfs attributes must call sysfs_attr_init() if
lockdep is enabled. In bman driver, we need invoke sysfs_attr_init()
for bman_dev_pool_count_attributes to avoid following call trace when
CONFIG_DEBUG_LOCK_ALLOC enabled:

fsl_b4xxx-wrs-linux/linux-windriver/3.10-r0/linux/kernel/lockdep.c:2987

[c0000000fb2a33c0] [c0000000000d8a18] .lockdep_init_map+0x2f8/0x6c0
[c0000000fb2a3490] [c000000000289fcc] .sysfs_add_file_mode+0xcc/0x150
[c0000000fb2a3540] [c00000000028e6f8] .internal_create_group+0x138/0x310
[c0000000fb2a35f0] [c000000000802808] .of_fsl_bman_probe+0x2a8/0x330

Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 drivers/staging/fsl_qbman/bman_config.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/staging/fsl_qbman/bman_config.c b/drivers/staging/fsl_qbman/bman_config.c
index 7b5bc88..53759a2 100644
--- a/drivers/staging/fsl_qbman/bman_config.c
+++ b/drivers/staging/fsl_qbman/bman_config.c
@@ -748,6 +748,7 @@ static int __devinit of_fsl_bman_probe(struct platform_device *ofdev)
 	}
 
 	for (i = 0; i < bman_pool_max; i++) {
+		sysfs_attr_init(&dev_attr_buffer_pool_count[i].attr);
 		ret = scnprintf((name_attrs_pool_count + i * 3), 3, "%d", i);
 		if (!ret)
 			goto del_group_4;
-- 
1.7.5.4

